src/lists.py

```diff
@@ -113,7 +117,31 @@ def reverse(x: DLList[T]) -> None:
     >>> print(x)
     [5, 4, 3, 2, 1]
     """
-    ...
+    link = x.head.next
+    end = x.head.prev
+    while link is not end:
+            insert_after(end,link.val)   #not the most efficient
+            remove_link(link)            # as we add one and delete one 
+            link = link.next             # but for now it works   
```

Comment:
You can certainly do it like that. I hadn't thought about it.

You can also just flip the references in each link:

```python
def reverse(x: DLList[T]) -> None:
    link = x.head
    while True:  # do...until (break at the end)
        link.prev, link.next = link.next, link.prev
        link = link.prev  # that was link.next a moment before
        if link is x.head: return
```